app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,221 @@
+    'insurance'                   => :keep,
+    'referred_to_clinic'          => :keep,
+    'resolved_without_fund'       => :keep,
+    'clinic_id'                   => :keep,
+    'urgent_flag'                 => :keep,
+    'procedure_cost'              => :keep,
+    'patient_contribution'        => :keep,
+    'naf_pledge'                  => :keep,
+    'fund_pledge'                 => :keep,
+    'pledge_sent'                 => :keep,
+    'pledge_generated_at'         => :keep,
+    'pledge_generated_by_id'      => :keep,
+  }.freeze
+
+
+  def has_alt_contact?
+    other_contact.present? || other_phone.present? || other_contact_relationship.present?
+  end
+
+  def set_age_range
+    case age
+    when nil, ''
+      :unknown
+    when 1..17
+      :under_18
+    when 18..24
+      :age18_24
+    when 25..34
+      :age25_34
+    when 35..44
+      :age35-44
+    when 45..54
+      :age45_54
+    when 55..100
+      :age55_100
+    else
+      'Bad value'
+    end
+  end
+
+  def shred(field)
+    self[field.to_sym] = nil
+  end
+
+  def empty(field)
+    self[field.to_sym] = []
+  end
+
+  def scramble(field)
+    self[field.to_sym] = SecureRandom.uuid
+  end
+
+  def overwrite(field)
+    self[field.to_sym] = "ARCHIVED"
+  end
+
+  def archive_fulfillment(field)
+    self.fulfillment.archive!
+  end
+
+  # TODO these archive calls are almost all identical
+  # on all the models. Abstract out somehow?
+  def archive_calls(field)
+    self.calls.each do |call|
+      call.archive!
+    end
+  end
+
+  def archive_notes(field)
+    self.notes.each do |note|
+      note.archive!
+    end
+  end
+
+  def archive_pledges(field)
+    self.external_pledges.each do |pledge|
+      pledge.archive!
+    end
+  end
+
+
+  def archive!
+    self.archived = true
+    self.had_other_contact = self.has_alt_contact?
+    self.age_range = set_age_range
+
+    self.attributes.keys.each do |field|
+      if PATIENT_ARCHIVE_REFERENCE.has_key?(field)
+        action = PATIENT_ARCHIVE_REFERENCE[field]
+        if action == :keep
+          #puts "Found a keep! (#{field})"
+          next
+        elsif self.respond_to?(action) # long term
+          #puts "Found a #{action} for field #{field}!"
+          self.send(action, field)
+        else
+          logger.warn "Func '#{action}' for field '#{field}' not found. Shredding"
+          #puts "Func '#{action}' for field '#{field}' not found. Shredding"
+          self.shred(field)
+        end
+      else
+        logger.warn "Found unaccounted for key '#{field}' when archiving. " \
+                    "Shredding the content of this field. Please add proper " \
+                    "handling for the field in the archive concern"
+        self.shred(field)
+      end
+    end
+    self.save
+  end
+
+  class_methods do
+    def unarchived
+      # TODO change to scope and acutally call this
+      Patient.where(archived: :false)
+    end
+
+    def archive
+      ready_to_archive do |patient|
```

Comment:
This is 100% my fault for not seeing this in the moment, but you want to slap an `each` on here to iterate through the collection. (I AM REALLY SORRY) 

The good news is that at that point it works!

```
Patient.archive
=> [#<Patient _id: 59d59444311c887197d67535, created_at: 2017-07-07 02:09:08 UTC, updated_at: 2017-10-05 02:09:54 UTC, user_ids: nil, clinic_id: BSON::ObjectId('59d59442311c887197d672a8'), pledge_generated_by_id: BSON::ObjectId('59d59440311c887197d6729c'), pledge_sent_by_id: nil, name: "ARCHIVED", primary_phone: nil, other_contact: nil, other_phone: nil, other_contact_relationship: nil, identifier: "8ef0bd20-4363-47fb-868e-07e1c05bf95a", voicemail_preference: "yes", line: "DC", language: "Spanish", initial_call_date: 2017-07-06 00:00:00 UTC, urgent_flag: true, last_menstrual_period_weeks: 6, last_menstrual_period_days: 5, age: nil, city: "Washington", state: "DC", county: "Washington", race_ethnicity: "Hispanic/Latino", employment_status: "Student", household_size_children: nil, household_size_adults: nil, insurance: "Other Insurance", income: "$10,000-14,999", special_circumstances: [], referred_by: nil, referred_to_clinic: true, appointment_date: 2017-08-05 00:00:00 UTC, procedure_cost: 555, patient_contribution: 120, naf_pledge: 120, fund_pledge: 115, pledge_sent: true, resolved_without_fund: false, pledge_generated_at: 2017-07-27 02:09:08 UTC, pledge_sent_at: nil, archived: true, age_range: "unknown", had_other_contact: false, version: 7, modifier_id: nil, created_by_id: BSON::ObjectId('59d59440311c887197d6729c'), updated_by_id: nil>]

Patient.where(name: 'ARCHIVED').count
=> 1
```

Reply:
`each` will be the death of me! 😆